lib/spack/spack/platforms/cray.py

```diff
@@ -61,7 +61,7 @@ def __init__(self):
                 _target = os.environ.get('SPACK_' + name.upper())
             if _target is None and name == 'back_end':
                 _target = self._default_target_from_env()
-            if _target is not None:
+            if _target:
```

Comment:
What real case is this change handling?

Reply:
`_default_target_from_env` can return whitespace.

Reply:
I think we should fix that in `_default_target_from_env` instead of here.

Reply:
I think this is clearer and more pythonic. I went back and forth on this.

`_default_target_from_env` runs a bash command and parses the output. If it gets anything, that's the target, if not, additional logic.

Reply:
I do not think `_default_target_from_env` should return whitespace for failure -- can it really return whitespace *or* `None`?

Reply:
It can't return `None` currently. It has to return a string.

Reply:
Reworked so it returns `None`, but I don't think this is an improvement.